Revise "information-theory_vn" - Phần 1 (#3081)
aivivn/d2l-vn: chapter_appendix-mathematics-for-deep-learning/information-theory_vn.md

In this section, we will investigate the fundamental concepts of information theory and applications of information theory in machine learning.
-->

> Chúng ta đang sống trong kỷ nguyên tràn ngập thông tin. 

Reviewer: ```suggestion
Vũ trụ mà chúng ta đang sống thì tràn ngập với thông tin. 
```

> Thông tin cung cấp một ngôn ngữ chung cho các lĩnh vực: từ thơ của Shakespeare đến các bài báo khoa học của các nhà nghiên cứu trên Cornell ArXiv,

Reviewer: ```suggestion
Thông tin cung cấp một ngôn ngữ chung vượt qua những vực thẳm phân tách nhiều lĩnh vực riêng biệt: từ thơ của Shakespeare đến các bài báo khoa học của các nhà nghiên cứu trên Cornell ArXiv,
```

Author: Câu này nghe complicated quá lkphuc ạ :).  Chữ 'vực thẳm phân tách' có lẽ nên sửa là ra 'hàng rào ngăn cách' cho nó nhẹ nhàng hơn chăng?

Reviewer: Tại bên tiếng anh có [hình ảnh](https://cdn.disclose.tv/sites/default/files/styles/og_image/public/img/post/2016/12/03/chasm-of-ignorance-physics-genius-reveals-how-everything-is-connected-136835.jpg?itok=12zpFuAJ) `chasm of ignorance` nên em dịch sát. Dịch thân quen với tiếng việt cũng được tuy bị mất đi hình ảnh này:
```suggestion
Thông tin cung cấp một ngôn ngữ chung giúp vượt qua rào cản ngăn cách nhiều lĩnh vực riêng biệt: từ thơ của Shakespeare đến các bài báo khoa học của các nhà nghiên cứu trên Cornell ArXiv,
```

-->

Trước khi bắt đầu, chúng ta hãy phác thảo mối quan hệ giữa học máy và lý thuyết thông tin.

> Học máy nhằm mục đích trích xuất các đặc trưng đáng chú ý từ dữ liệu và đưa ra các dự đoán quan trọng.

Reviewer: ```suggestion
Mục tiêu của học máy là trích xuất các đặc trưng đáng chú ý từ dữ liệu và đưa ra các dự đoán quan trọng.
```

> Mặt khác, lý thuyết thông tin nghiên cứu mã hóa, giải mã, truyền và thao tác thông tin.

Reviewer: ```suggestion
Mặt khác, lý thuyết thông tin nghiên cứu vấn đề mã hóa, giải mã, truyền và thao tác với thông tin.
```

> Kết quả là, lý thuyết thông tin cung cấp ngôn ngữ cơ bản để thảo luận về việc xử lý thông tin trong các hệ thống học máy.

Reviewer: ```suggestion
Do vậy, lý thuyết thông tin cung cấp ngôn ngữ cơ bản để thảo luận về việc xử lý thông tin trong các hệ thống học máy.
```

Ví dụ: nhiều ứng dụng học máy sử dụng mất mát entropy chéo như được mô tả trong :numref:`sec_softmax`.

> Mất mát này có thể trực tiếp rút ra từ các quan điểm lý thuyết thông tin.

Reviewer: ```suggestion
Mất mát này có thể được trực tiếp rút ra từ các quan điểm từ góc nhìn lý thuyết thông tin.
```

-->

Ta hãy bắt đầu với "linh hồn" của lý thuyết thông tin: thông tin.
*Thông tin* có thể được mã hóa trong bất kỳ thứ gì với một hoặc nhiều chuỗi định dạng mã hóa.

> Giả sử chính chúng ta đề ra nhiệm vụ thử định nghĩa một khái niệm thông tin.

Reviewer: ```suggestion
Ta giả sử thử chính mình đưa ra một định nghĩa cho khái niệm thông tin.
```

> Điểm chúng ta có thể bắt đầu là gì?

Reviewer: ```suggestion
Ta có thể bắt đầu từ đâu?
```

Đầu tiên, họ lật một lá và nói, "Tôi thấy một lá bài."
Điều này không cung cấp cho ta thông tin nào.

> Nên chúng ta rõ ràng chắc chắn rằng trong trường hợp này thông tin sẽ là 0.

Reviewer: ```suggestion
Rõ ràng là trong trường hợp này chúng ta đã biết chắc chắn mệnh đề trên là đúng nên lượng thông tin mà nó chứa sẽ là 0.
```

-->

Tiếp theo, họ lật một lá khác và nói, "Tôi thấy một lá cơ."
Điều này cung cấp cho ta một chút thông tin, mà trên thực tế chỉ có thể có $4$ loại chất khác nhau , mỗi chất đều có khả năng như nhau, vì vậy ta không ngạc nhiên trước kết quả này.

> Ta hy vọng rằng với bất kể hình thức đo đạc thông tin nào, sự kiện này nên có hàm lượng thông tin thấp.

Reviewer: ```suggestion
Ta hy vọng rằng dù thông tin được đo đạc dưới bất kể hình thức nào, sự kiện này nên có hàm lượng thông tin thấp.
```

Indeed there were $52$ equally likely possible outcomes, and our friend told us which one it was. This should be a medium amount of information.
-->

> Tiếp theo, họ lật một lá và nói, "Đây là quân $3$ bích.". Đây là thông tin thêm.

Reviewer: ```suggestion
Tiếp theo, họ lật một lá và nói, "Đây là quân $3$ bích.". Câu trên chứa nhiều thông tin hơn.
```

-->

Hãy đi đến cực hạn.
Giả sử rằng cuối cùng họ lật từng lá bài từ bộ bài và đọc ra toàn bộ trình tự của bộ bài đã bị xáo trộn đó.

> Có $52!$ các thứ tự khác nhau cho bộ bài, một lần nữa tất cả đều có khả năng như nhau, vì vậy chúng ta cần rất nhiều thông tin để biết trình từ nào của việc rút bài xảy ra.

Reviewer: ```suggestion
Có $52!$ các thứ tự khác nhau cho bộ bài với khả năng như nhau, vì vậy chúng ta cần rất nhiều thông tin để biết được chính xác trình tự rút bài.
```

-->

Bất kỳ khái niệm thông tin nào chúng ta phát triển phải phù hợp với trực giác này.

> Thật vậy, trong phần tiếp theo, chúng ta sẽ học cách tính toán rằng các sự kiện trên có $0\text{ bit}$,
> $2\text{ bit}$, $~5.7\text{ bit}$, và $~225.6\text{ bit}$ của thông tin tương ứng.

Reviewer: ```suggestion
Thật vậy, trong phần tiếp theo, chúng ta sẽ học cách tính toán rằng các sự kiện trên có tương ứng $0\text{ bit}$,
$2\text{ bit}$, $~5.7\text{ bit}$, và $~225.6\text{ bit}$ thông tin.
```

For a common event, we may not need much information.
-->

Nếu đọc hết những thí nghiệm tưởng tượng này, chúng ta thấy một ý tưởng tự nhiên.

> Như một điểm khởi đầu, thay vì quan tâm đến điều đã biết,

Reviewer: ```suggestion
Để khởi đầu, thay vì quan tâm đến kiến thức đã biết,
```

chúng ta có thể xây dựng ý tưởng là thông tin đại diện cho mức độ bất ngờ hoặc xác suất trừu tượng của sự kiện.
Ví dụ, nếu chúng ta muốn mô tả một sự kiện hiếm gặp, chúng ta cần rất nhiều thông tin.

> Đối với một sự kiện thông thường, chúng ta có thể không cần nhiều thông tin.

Reviewer: ```suggestion
Đối với một sự kiện phổ biến, chúng ta có thể không cần nhiều thông tin.
```

In his article, Shannon introduced the concept of information entropy for the first time. We will begin our journey here.
-->

> Năm 1948, Claude E. Shannon xuất bản *Lý thuyết Toán cho Truyền tin - A Mathematical Theory of Communication* :cite:`Shannon.1948` thiết lập lý thuyết thông tin.

Reviewer: ```suggestion
Năm 1948, Claude E. Shannon thiết lập lĩnh vực lý thuyết thông tin qua bài báo khoa học *Lý thuyết Toán cho Truyền tải Thông tin - A Mathematical Theory of Communication* :cite:`Shannon.1948` .
```

And hence, a series of binary digits of length $n$ contains $n$ bits of information.
-->

Vì thông tin biểu diễn xác suất trừu tượng của một sự kiện, làm thế nào để chúng ta ánh xạ xác suất đó thành số lượng bit?

> Shannon đã giới thiệu thuật ngữ *bit* làm đơn vị thông tin, thuật ngữ này ban đầu do John Tukey tạo ra.

Reviewer: ```suggestion
Shannon đã giới thiệu thuật ngữ *bit* làm đơn vị thông tin, mà ban đầu được đề xuất bởi John Tukey.
```

> Vậy "bit" là gì và tại sao ta sử dụng nó để đo lường thông tin? Trong lịch sử, một máy phát cổ chỉ có thể gửi hoặc nhận hai loại mã: $0$ và $1$.

Reviewer: ```suggestion
Vậy "bit" là gì và tại sao ta sử dụng nó để đo lường thông tin? Trong quá khứ, một máy phát tín hiệu chỉ có thể gửi hoặc nhận hai loại mã: $0$ và $1$.
```

> Thật vậy, mã hóa nhị phân vẫn được sử dụng phổ biến trên tất cả các máy tính kỹ thuật số hiện đại.

Reviewer: ```suggestion
Mà thật ra mã hóa nhị phân vẫn được sử dụng phổ biến trên tất cả các máy tính kỹ thuật số hiện đại.
```

Bằng cách này, bất kỳ thông tin nào cũng được mã hóa bởi một chuỗi $0$ và $1$.

> Và do đó, một chuỗi các chữ số nhị phân có độ dài $n$ chứa $n$ bit thông tin.

Reviewer: ```suggestion
Và do đó, một chuỗi các chữ số nhị phân (*binary*) có độ dài $n$ chứa $n$ bit thông tin.
```
Thêm chú thích tiếng anh do chỗ này đang giải thích nguồn gốc thuật ngữ.

For example, the code "0010" has a self-information
-->

> như là số *bit* thông tin ta đã nhận cho sự kiện $X$ này. 

Reviewer: ```suggestion
là số *bit* thông tin ta đã nhận cho sự kiện $X$ này. 
```

Before that, let us first import all the necessary packages in this section.
-->

> Chúng ta có thể tính toán lượng tin như hình dưới đây. 

Reviewer: ```suggestion
Chúng ta có thể tính toán lượng tin như phần dưới đây.
```
